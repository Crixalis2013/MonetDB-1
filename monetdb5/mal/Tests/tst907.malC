function one(base:int):int;
	#io.print(base);
	return base;
end one;

function three(base:int,x:int,y:int):int;
	#io.print(base);
	return base;
end three;

function pow(base:int,raise:int):int;
        mg_a := calc.dbl(base);
        ng_a := calc.dbl(raise);
        og_a := calc.*(mg_a,ng_a);
        pg_a := calc.int(og_a);
        return pg_a;
end pow;

function foo():bat[:int];
k:= user.one(1);
l:= user.three(1,2,3);
p:= user.pow(1,2);

b:= bat.new(:int);
bat.append(b,1);

io.printf("[ do multiplex three test, ");
z:bat[:int]:=mal.multiplex("user","three",b,2,3);
io.printf("do multiplex one test, ");
z:bat[:int]:=mal.multiplex("user","one",b);
io.printf("do multiplex pow test ]\n");
z:bat[:int]:=mal.multiplex("user","pow",b,2);
return z;
end;

optimizer.multiplex("user","foo");

z := user.foo();
io.print(z);
