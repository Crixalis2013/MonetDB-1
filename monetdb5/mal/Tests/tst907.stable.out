stdout of test 'tst907` in directory 'monetdb5/mal` itself:


# 08:01:38 >  
# 08:01:38 >  Mtimeout -timeout 60 Mserver "--config=/ufs/mk/monet5/Linux/etc/MonetDB5.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5/Linux/lib/MonetDB5:/ufs/mk/opensource/MonetDB/Linux/lib/MonetDB" --set "gdk_dbfarm=/ufs/mk/monet5/Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/monet5/Linux/var/MonetDB5/log" --set mapi_port=42158 --set sql_port=55708 --set monet_prompt= --trace "--config=/ufs/mk/monet5/src/mal/Tests/All.conf" --dbname=mTests_src_mal  tst907.mal </dev/null
# 08:01:38 >  

[ "do multiplex three test",	"do multiplex one test",	"do multiplex pow test"	]
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	  2	  ]

# 08:01:38 >  
# 08:01:38 >  Done.
# 08:01:38 >  

